protos/feast/core/FeatureService.proto

```diff
@@ -25,7 +30,13 @@ message FeatureServiceSpec {
 
   // List of features that this feature service encapsulates.
   // Stored as a list of references to other features views and the features from those views.
-  repeated FeatureViewProjection features = 3;
+  repeated string features = 6;
+
+  repeated FeatureTable feature_tables = 7;
```

Comment:
I don't understand what these new objects are supposed to contain. Don't we just need `features`?

Reply:
FeatureViews and OnDemandFeatureViews will carry the info of the modified fields such as modified name. I also added a field for FeatureTable for consistency.

Reply:
Looking at this now, it doesn't look like the right approach. Wouldn't we run into consistency problems if we store copies of feature view objects with different configurations?

The original problem was 
> The trouble right now is that at query time, even though the FV is one concept in the users mind when use get_historical_features or get_online_features, but it actually splits into a different class when the getitem magic method is triggered by my_fv[[<my-feature-selections]] and it becomes a FVProjection. This means if users wanted to transform their FV, we'd have to support these methods on both the FV class and FVProjection class.

This is both a user facing API problem and a code duplication problem, but I am beginning to think that having an object like a FeatureViewProjection within the protos might be the right approach. Assuming the user could modify a FeatureView at their own whims, and assuming we could convert that final feature view into something like

```proto
message FeatureViewProjection {
  string feature_view_name = 1;
  repeated FeatureSpecV2 feature_columns = 2;
  string feature_view_name_alias = 3;
  something entity_mapping_etc = 4;
}
```

Wouldn't that both solve the consistency problem in the registry (normalized instead of denormalized), as well as the user facing API problem where we have to maintain FeatureViewProjections?